cfme/v2v/infrastructure_mapping.py

```diff
@@ -0,0 +1,329 @@
+                            ...
+                        }
+            """
+            for cluster in values if isinstance(values, dict) else {}:
+                if self.cluster_selector.is_displayed:
+                    self.cluster_selector.fill(cluster)
+                for mapping in values[cluster]["mappings"]:
+                    self.source_networks.fill(mapping["sources"])
+                    self.target_networks.fill(mapping["target"])
+                    self.add_mapping.click()
+            was_change = not self.mappings_tree.is_empty
+            self.after_fill(was_change)
+            return was_change
+
+    @View.nested
+    class result(View):  # noqa
+        close = Button("Close")
+        continue_to_plan_wizard = Button("Continue to the plan wizard")
+        fill_strategy = WaitFillViewStrategy()
+
+        @property
+        def is_displayed(self):
+            return self.continue_to_plan_wizard.is_displayed
+
+    def after_fill(self, was_change):
+        if was_change:
+            self.result.close.click()
+
+
+class AllInfraMappingView(InfrastructureMappingView):
+    """All infrastructure mapping View"""
+
+    infra_mapping_list = InfraMappingList("infra-mappings-list-view")
+    create_infra_mapping = Text(locator='(//a|//button)[text()="Create Infrastructure Mapping"]')
+    configure_providers = Text(locator='//a[text()="Configure Providers"]')
+    paginator_view = View.include(V2VPaginatorPane, use_parent=True)
+    search_box = TextInput(locator=".//div[contains(@class,'input-group')]/input")
+    clear_filters = Text(".//a[text()='Clear All Filters']")
+    sort_order = Text(".//button[./span[contains(@class,'sort-direction')]]")
+    filter_by_dropdown = SelectorDropdown("id", "filterFieldTypeMenu")
+    sort_by_dropdown = SelectorDropdown("id", "sortTypeMenu")
+
+    @property
+    def is_displayed(self):
+
+        return len(self.browser.elements(".//div[contains(@class,'spinner')]")) == 0 and (
+            self.create_infra_mapping.is_displayed
+            or self.infra_mapping_list.is_displayed
+            or self.configure_providers.is_displayed
+        )
+
+
+class AddInfrastructureMappingView(InfrastructureMappingForm):
+    form_title_text = VersionPicker(
+        {
+            Version.lowest(): "Infrastructure Mapping Wizard",
+            "5.10": "Create Infrastructure Mapping",
+        }
+    )
+
+    @property
+    def is_displayed(self):
+        return self.title.text == self.form_title_text
+
+
+class EditInfrastructureMappingView(InfrastructureMappingForm):
+    form_title_text = "Edit Infrastructure Mapping"
+    save = Button("Save")
+
+    @property
+    def is_displayed(self):
+        return self.title.text == self.form_title_text
+
+
+@attr.s
+class InfrastructureMapping(BaseEntity, Updateable):
+    """Class representing v2v infrastructure mappings"""
+
+    name = attr.ib()
+    description = attr.ib(default=None)
+    form_data = attr.ib(default=None)
```

Comment:
Ultimately I think that this needs to be broken down programatically into attributes and potentially wrapper classes for things like cluster mappings. An `InfrastructureMapping` instance should have `name`, `description`, `source_cluster`, `target_cluster`, etc. with a `fill_values` method that generates a data structure matching the view structure.  Explicit mapping attributes over a loose 'form_data' definition that is broken up across several view class docblocks above. Looking at this class, its completely opaque what this is supposed to look like.

Reply:
I can retrieve fields from form_data and fill each view separately but that work is being done in each fil.
Let me know if you want me to make these changes .